src/persist-client/src/batch.rs

```diff
@@ -206,6 +210,56 @@ impl From<&PersistConfig> for BatchBuilderConfig {
 /// then finishing it.
 #[derive(Debug)]
 pub struct BatchBuilder<K, V, T, D>
+where
+    K: Codec,
+    V: Codec,
+    T: Timestamp + Lattice + Codec64,
+{
+    pub(crate) stats_schemas: Schemas<K, V>,
+    pub(crate) builder: BatchBuilderInternal<K, V, T, D>,
```

Comment:
yeah, it's the worst. this is because compaction needs them to be different: it "decodes" them using the `Vec<u8>` Codec but the stats obviously need to be generated via the real Codec/Schema. a better way to do this would be for compaction to operate on the real K and V (which is potentially desirable anyway for consolidation), but that's certainly not something I want to pick off as part of this PR

Reply:
Mmm... so it's not that they're ever expected to disagree _when we're using BatchBuilder_, but the internal batch builder needs to take stats as a parameter, so we need to keep around an extra copy to pass in.

And yeah, as a temporary thing seems fine to punt!

Reply:
> Mmm... so it's not that they're ever expected to disagree when we're using BatchBuilder, but the internal batch builder needs to take stats as a parameter, so we need to keep around an extra copy to pass in.

yeah, that's exactly correct. forgot to include this context (which you have clearly now figured out), my initial attempt was to add new methods to BatchBuilder that took the stats_schema as a parameter, but that hits ownership issues that I think are unresolvable, which is how I wound up here